website/static/js/pages/file-page.js

```diff
@@ -58,4 +58,52 @@ $(function() {
         });
     }
 
+    var titleEditable = function () {
+        var readOnlyProviders = ['bitbucket', 'figshare', 'dataverse'];
+        var ctx = window.contextVars;
+        if (readOnlyProviders.includes(ctx.file.provider) || ctx.file.checkoutUser || !ctx.currentUser.canEdit || ctx.node.isRegistration)
```

Comment:
@alexschiller Is this all correct? What if the checkoutUser is the current user? Can they then check out the file? Also, does the primary preprint file also check out the file, or is that considered a bit different?

Reply:
On the plus side, it'll still error even if we aren't catching all of the conditions, but I think that the current checkout user may worth checking as possibly overly aggressive. I believe the user who has currently checked out the file can edit it.

Reply:
@brianjgeiger Looks right to me-- Via fangorn, the only two actions allowed for a checkout user are uploading a new version and checking in the file. [The rename button does not show up if the file is checked out.](https://github.com/CenterForOpenScience/osf.io/blob/develop/website/static/js/fangorn.js#L1918)

The configs for bitbucket, figshare, and dataverse don't write a rename button at all. 

So all in all, AFAICT this matches fangorn functionality. Whether the fangorn behavior is desirable or not is questionable.